Decorator/TextDecorator/EmojiMessageDecorator.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+namespace Decorator
+{
+    public class EmojiDecorator : MessageDecorator
+    {
+        public EmojiDecorator(IMessage message) : base(message)
+        { }
+
+        public override string GetMessage()
+        {
+            string text = message.GetMessage();
```

Comment:
You reveal internal implementation of the base class using directly `message`. It's better to use `base` instead

Reply:
done